### ISSUE TITLE
uNet results not matching expected results

### ISSUE THREAD

**Issue by S4WRXTTCS** [qubvel/segmentation_models#93]
I really like this Library from the perspective of being really flexible so I can train a uNet model, or an FCN model, etc. Where it can be a single class, or multi-class.

But, I'm having difficulties matching the results I got using a much more limited uNet example that only supported single class (or 2 if you count the background) datasets.

Here is the results I got from the much more limited Unet example. 
![result_other](https://user-images.githubusercontent.com/280576/57041884-278bd080-6c18-11e9-9503-5d470226c77b.png)

Here is the results I'm getting using this library.
![result](https://user-images.githubusercontent.com/280576/57042072-9406cf80-6c18-11e9-9ff5-4b742ccf6808.png)

I'm not a coder so my code is really just a copy/paste job of things I found while browsing the documentation for this repository, and the closed/open issues to figure out how to load the data for the model/etc.

So it's likely that I missed something. 

My code
[Code.zip](https://github.com/qubvel/segmentation_models/files/3135955/Code.zip)

seg_train.py trains the model, and saves the model. It gives me some really low loss numbers on the validation data (around .02), and really high IoU numbers (.95 or so). So I'm really confused as to why the results are so bad. 

seg_test.py loads the models, and does inference on a single image. I've tested seg_test.py with the uNet model I previously trained using code from a different github repository, and it seems to work fine. I initially suspected something was wrong with this code, but that doesn't seem to be the case. 

The Dataset I used
https://drive.google.com/open?id=1T1i_v3ApSDPPiy4TAUmdQ4VJNXPc30jW

**Comment by qubvel** [comment]
Hi @S4WRXTTCS 
I looked quickly at your code:  
First of all, you use different ways to read images:
 - skimage.imread
 - keras.preprocessing.image  

Second, after resizing image can be converted from uint8 -> float32 (0..255 -> 0..1 range respectively).
Check that before using `preprocessing_fn` your `x` is uint8 data (0..255 range) both during training and during testing.   

This should help you, good luck!

**Comment by S4WRXTTCS** [comment]
Thanks. 

For some reason I thought I had accounted for that because the code I copy/pasted was from issue #74, and I knew that I had have account for the resize turning uint8 images into float64.  But, somehow I forgot to do that. 

With that change I get results that look pretty good.

![result](https://user-images.githubusercontent.com/280576/57046140-5ad45c80-6c24-11e9-8286-2a98fc34e2f5.png)

Here is the code with the change that was commented about in Issue #74, but I don't have the issue the person had with my validation dataset.  Everything seems as expected, but there still might be something wrong with it. 

[Code.zip](https://github.com/qubvel/segmentation_models/files/3136179/Code.zip)
